src/js/media-controller.js

```diff
@@ -152,18 +154,37 @@ class MediaController extends MediaContainer {
       this.propagateMediaState(MediaUIProps.MEDIA_STREAM_TYPE);
 
     } else if (attrName === Attributes.FULLSCREEN_ELEMENT) {
-      const el = newValue
-        // @ts-ignore
-        ? this.getRootNode()?.getElementById(newValue)
-        : undefined;
+      const el = newValue ? this.#rootNode?.getElementById(newValue) : undefined;
       // NOTE: Setting the internal private prop here to not
       // clear the attribute that was just set (CJP).
       this.#fullscreenElement = el;
     }
+  }
 
-    super.attributeChangedCallback(attrName, oldValue, newValue);
+  connectedCallback() {
+    // mediaSetCallback() is called in super.connectedCallback();
+    super.connectedCallback();
+
+    // getRootNode() in disconnectedCallback returns the media-controller element itself
+    // but we need the HTMLDocument or ShadowRoot if media-controller is in a shadow DOM.
+    // We store the correct root node here so we can access it later.
+    this.#rootNode = /** @type HTMLDocument | ShadowRoot */ (this.getRootNode());
+
+    this.enableHotkeys();
+  }
+
+  disconnectedCallback() {
+    // mediaUnsetCallback() is called in super.disconnectedCallback();
+    super.disconnectedCallback();
+    this.disableHotkeys();
+    // Disable captions on disconnect to prevent a memory leak if they stay enabled.
+    toggleSubsCaps(this, false);
```

Comment:
this fixed a 2nd memory leak, I don't know the specifics but when subtitles stay enabled the detached video element doesn't get GC'd

Reply:
@luwes in flight PR #770 removes the `toggleSubsCaps()` utility method. Can you refactor to something like this instead?
https://github.com/muxinc/media-chrome/pull/770/files/681fe4d8cb5fc323b663f52de67a77fa2351b7fa#diff-2b43f6e6b30d1c84504e425bd4fcaa330513d0389aed6f8127ecef8bbae96629R138-R147